### PR TITLE
Adding XCFA support to Theta

### PR DESCRIPTION
**Pull request by leventeBajczi**

### CODE REVIEW

File: subprojects/core/src/main/java/hu/bme/mit/theta/core/clock/op/ClockOps.java [hunk 1]
```diff
@@ -15,18 +15,11 @@
...
 import hu.bme.mit.theta.core.clock.constr.ClockConstrs;
 import hu.bme.mit.theta.core.decl.Decl;
 import hu.bme.mit.theta.core.decl.VarDecl;
-import hu.bme.mit.theta.core.stmt.AssignStmt;
-import hu.bme.mit.theta.core.stmt.AssumeStmt;
-import hu.bme.mit.theta.core.stmt.HavocStmt;
-import hu.bme.mit.theta.core.stmt.SkipStmt;
-import hu.bme.mit.theta.core.stmt.Stmt;
-import hu.bme.mit.theta.core.stmt.StmtVisitor;
+import hu.bme.mit.theta.core.stmt.*;
```
Comment (hajduakos) [inline subprojects/core/src/main/java/hu/bme/mit/theta/core/clock/op/ClockOps.java]: We prefer not to use `import *`

File: subprojects/core/src/main/java/hu/bme/mit/theta/core/dsl/CoreDslManager.java [hunk 2]
```diff
@@ -15,24 +15,20 @@
...
 import hu.bme.mit.theta.common.dsl.BasicScope;
 import hu.bme.mit.theta.core.decl.Decl;
 import hu.bme.mit.theta.core.dsl.gen.CoreDslLexer;
 import hu.bme.mit.theta.core.dsl.gen.CoreDslParser;
-import hu.bme.mit.theta.core.dsl.impl.ExprCreatorVisitor;
-import hu.bme.mit.theta.core.dsl.impl.ExprWriter;
-import hu.bme.mit.theta.core.dsl.impl.StmtCreatorVisitor;
-import hu.bme.mit.theta.core.dsl.impl.StmtWriter;
-import hu.bme.mit.theta.core.dsl.impl.TypeCreatorVisitor;
+import hu.bme.mit.theta.core.dsl.impl.*;
```
Comment (hajduakos) [inline subprojects/core/src/main/java/hu/bme/mit/theta/core/dsl/CoreDslManager.java]: We prefer not to use `import *`

File: subprojects/core/src/main/java/hu/bme/mit/theta/core/dsl/impl/StmtWriter.java [hunk 3]
```diff
@@ -15,11 +15,7 @@
  */
 package hu.bme.mit.theta.core.dsl.impl;
 
-import hu.bme.mit.theta.core.stmt.AssignStmt;
-import hu.bme.mit.theta.core.stmt.AssumeStmt;
-import hu.bme.mit.theta.core.stmt.HavocStmt;
-import hu.bme.mit.theta.core.stmt.SkipStmt;
-import hu.bme.mit.theta.core.stmt.StmtVisitor;
+import hu.bme.mit.theta.core.stmt.*;
```
Comment (hajduakos) [inline subprojects/core/src/main/java/hu/bme/mit/theta/core/dsl/impl/StmtWriter.java]: We prefer not to use `import *`. I will not add this comment, but should be fixed in other places.

File: subprojects/core/src/main/java/hu/bme/mit/theta/core/stmt/xcfa/XcfaStmtVisitor.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ */
+package hu.bme.mit.theta.core.stmt.xcfa;
+
+import hu.bme.mit.theta.core.stmt.StmtVisitor;
+
+public interface XcfaStmtVisitor<P, R> extends StmtVisitor<P, R> {
+
+	R visit(XcfaCallStmt stmt, P param);
+
+    R visit(StoreStmt storeStmt, P param);
```
Comment (hajduakos) [inline subprojects/core/src/main/java/hu/bme/mit/theta/core/stmt/xcfa/XcfaStmtVisitor.java]: Inconsistent formatting

**Comment by leventeBajczi** [comment]
Resolved wildcard imports and inconsistent formatting. Wildcard imports were present in other parts of the project as well, replaced them with specific imports as well - now there are no `import *` lines in the project.

**Pull request merged**